### PR TITLE
AVRO-3914: Add nanos support for the Java SDK

### PR DESCRIPTION
**Pull request by Fokko**
## What is the purpose of the change

[AVRO-3914](https://issues.apache.org/jira/browse/AVRO-3914)

## Verifying this change

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
- *Extended interop tests to verify consistent valid schema names between SDKs*
- *Added test that validates that Java throws an AvroRuntimeException on invalid binary data*
- *Manually verified the change by building the website and checking the new redirect*

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by clesaec** [comment]
May need a doc change [on time description](https://avro.apache.org/docs/1.11.1/specification/#time-millisecond-precision)
and a JIRA would also be welcomed (to trace feature and comments).
Other than that, code proposal is nice.

**Comment by Fokko** [comment]
Hey @clesaec thanks for the comment, the doc change can be found here: https://github.com/apache/avro/pull/2554

**Review by clesaec — APPROVED** [review]
LGTM
(to keep link with doc change : https://github.com/apache/avro/pull/2554)

File: lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java [hunk 1]
```diff
@@ -267,6 +281,12 @@ public static LocalTimestampMicros localTimestampMicros() {
     return LOCAL_TIMESTAMP_MICROS_TYPE;
   }
 
+  private static final LocalTimestampMicros LOCAL_TIMESTAMP_NANOS_TYPE = new LocalTimestampMicros();
+
+  public static LocalTimestampMicros localTimestampNanos() {
+    return LOCAL_TIMESTAMP_NANOS_TYPE;
+  }
+
```
Comment (arkanovicz) [inline lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java:289]: Shoudln't it be LocalTimestampNanos class here?

Comment (Fokko) [reply lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java:289]: Yes, you're correct! Do you want to raise the PR?

Comment (arkanovicz) [reply lang/java/avro/src/main/java/org/apache/avro/LogicalTypes.java:289]: Done at https://github.com/apache/avro/pull/2706

**Pull request merged**